### ISSUE TITLE
Server-alias doesn't not work on http:80 => 0.7.2 to 0.8.3

### ISSUE THREAD

**Issue by AlexisDuf** [jcmoraisjr/haproxy-ingress#565]
Hello, 

I tried to upgrade from 0.7.2 to 0.8.3 but it seems I'm facing to some issues when I use `server-alias` annotations. 

I'm currently run kubernetes on 1.16 version. 
My main domain is *.k8s-hpr.my.domain.fr
For one of my app the subdomain myapp.k8s-hpr.my.domain.fr and I created a server alias myapp.domain.fr

It was working perfectly on 0.7.2 but with 0.8.5 is not working anymore. 

After digging in the config file of the ingress controller I noticed some strange things. (By the way I'm not an expert at all of Haproxy conf) 

Problems seems to be on map file, I did a grep on the map to check if I can find the server alias. 
```bash
grep 'myapp' /etc/haproxy/maps/_*
/etc/haproxy/maps/_front001_host.map:myapp.k8s-hpr.my.domain.fr/ myapp_occ-service-integration_8080
/etc/haproxy/maps/_front001_host.map:myapp.domain.fr/ myapp_occ-service-integration_8080
/etc/haproxy/maps/_global_http_front.map:myapp.k8s-hpr.my.domain.fr/ myapp_occ-service-integration_8080
/etc/haproxy/maps/_global_https_redir.map:myapp.k8s-hpr.my.domain.fr/ no
/etc/haproxy/maps/_global_k8s_ns.map:myapp.k8s-hpr.my.domain.fr/ -
/etc/haproxy/maps/_public.list:myapp.k8s-hpr.my.domain.fr
```
According to me we should have:
```bash
myapp.domain.fr/ myapp_occ-service-integration_8080
``` 
In `/etc/haproxy/maps/_global_http_front.map` 

Because based the `haproxy.cfg` for http without https redirect we use `_global_http_front.map` map conf
```
haproxy.cfg
...
...
frontend _front_http
    mode http
    bind *:80
    http-request set-var(req.base) base,lower,regsub(:[0-9]+/,/)
    http-request redirect scheme https if { var(req.base),map_beg(/etc/haproxy/maps/_global_https_redir.map,_nomatch) yes }
....
    http-request set-var(req.backend) var(req.base),map_beg(/etc/haproxy/maps/_global_http_front.map,_nomatch)
    use_backend %[var(req.backend)] unless { var(req.backend) _nomatch }
    default_backend _default_backend
```

So according to my understanding the value of `req.backend` will always be `_nomatch` :)

**Comment by jcmoraisjr** [comment]
Hi, thanks for the detailed report! Can you confirm https works as expected?

The following patch (against `release-0.8`) should solve this issue, perhaps you can give it a chance. Otherwise I'll have a look on it myself and release a new v0.8 within a couple of days.

```diff
diff --git a/pkg/haproxy/config.go b/pkg/haproxy/config.go
index 9dacb78..e7d3dcf 100644
--- a/pkg/haproxy/config.go
+++ b/pkg/haproxy/config.go
@@ -385,6 +385,8 @@ func (c *config) BuildFrontendGroup() error {
 				}
 				if !hasSSLRedirect || c.global.Bind.HasFrontingProxy() {
 					fgroup.HTTPFrontsMap.AppendHostname(base, back)
+					fgroup.HTTPFrontsMap.AppendAliasName(aliasName, back)
+					fgroup.HTTPFrontsMap.AppendAliasRegex(aliasRegex, back)
 				}
 				var ns string
 				if host.VarNamespace {
```

**Comment by AlexisDuf** [comment]
Hi Joao, 

Unfortunately I'm not able to check on https if it's working well. https is handle by physical HAP in front of my ingresses. 

I will try to do a custom build on my side with the change you recommend and deploy it on my clusters then I will let you know if it works or not :)

If I have time i will propose you a PR 👍

**Comment by jcmoraisjr** [comment]
Hi thanks! A local test would be great. Note that 1) if building your own copy, v0.8 uses go1.9.2; 2) is issuing a PR, send against master, I'll send the cherry pick to the release branches; and finally 3) `config.go` changed a bit from v0.8 to v0.11 but the fix is pretty much the same, just using another local var.

**Comment by AlexisDuf** [comment]
Hi @jcmoraisjr 

I tested the change, and it works on my side.
I forked the project and did the change, but not sure which test need to be updated for that case. Will you take care of it ?

**Comment by jcmoraisjr** [comment]
Hi, great, thanks. Please send your proposal into a new PR so we can see what will break and how to fix.

**Comment by jcmoraisjr** [comment]
Too fast, the diff is just here. I'll have a look and push this fix. Thanks.

**Comment by jcmoraisjr** [comment]
Fixed on v0.8.4, v0.9 and v0.10-beta.1. Closing.
